cli/flox-rust-sdk/src/providers/build.rs

```diff
@@ -3077,6 +3077,108 @@ mod tests {
     async fn build_result_only_has_runtime_packages_sandbox_pure() {
         build_result_only_has_runtime_packages(true).await;
     }
+
+    /// Test that cmake can make use of the CMAKE_PREFIX_PATH variable as
+    /// set by etc-profiles.
+    async fn build_can_use_cmake(sandbox: bool) {
+        let package_name = String::from("hello-cmake");
+        // from: test_data/input_data/build/hello-cmake/HelloTarget/share/hello/HelloTarget.cmake
+        let file_name = String::from("hello.txt");
+        // from: test_data/input_data/build/hello-cmake/hello.in
+        let file_content = String::from("Hello, world!\n");
+
+        // Start by initializing a flox instance and tmpdir.
+        let (mut flox, _temp_dir_handle) = flox_instance();
+
+        // We have to materialize the environment at test time because of the
+        // requirement to install by storepath, so we start copying in the
+        // build environment which is comprised of source code and an empty
+        // flox environment.
+        let path = GENERATED_DATA.join("build/hello-cmake");
+        let path: &Path = path.as_ref();
+        copy_dir_recursive(path, &flox.temp_dir, true).unwrap();
+
+        // Import the HelloTarget directory as a package to be installed.
```

Comment:
suggestion blocking: I might not be understanding exactly why we need to add HelloTarget as a store path, but I think we should try to emulate what a user would do in a project that uses CMake. Could we test this behavior without the `nix store` invocation? Happy to pair or push a change if you're running into issues with the mocks and stuff